File: Pandacap.LowLevel/Data/Credentials/WeasylCredentials.fs
namespace Pandacap.Data

open System.ComponentModel.DataAnnotations
open Pandacap.PlatformBadges

/// The active credentials for Weasyl.
type WeasylCredentials() =
    [<Key>]
    member val Login = "" with get, set
    member val ApiKey = "" with get, set

    interface IExternalCredentials with
        member this.Username = this.Login
        member _.Platform = Weasyl

File: Pandacap.Database/WeasylCredentials.cs
using Pandacap.UI.Badges;
using Pandacap.UI.Elements;
using System.ComponentModel.DataAnnotations;

namespace Pandacap.Database
{
    public class WeasylCredentials : IExternalCredentials
    {
        [Key]
        public string Login { get; set; } = "";

        public string ApiKey { get; set; } = "";

        Badge IExternalCredentials.Badge => Badges.Weasyl;

        string IExternalCredentials.Username => Login;

        string IExternalCredentials.PlatformName => "Weasyl";
    }
}
